src/DynamoCore/Graph/Nodes/NodeInputData.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Graph.Nodes
+{
+    public enum NodeInputTypes
+    {
+        numberInput, booleanInput, stringInput, colorInput, dateInput, selectionInput
+    };
+
+
+
+    /// <summary>
+    /// Represents a node which acts as a UI input for the graph
+    /// - may also hold a value for that input
+    /// </summary>
+    public class NodeInputData
+    {
+        /// <summary>
+        /// The id of the node.
+        /// </summary>
+        public string Id { get; set; }
+        /// <summary>
+        /// Display name of the input node.
+        /// </summary>
+        public string Name { get; set; }
+        /// <summary>
+        /// The type of input this node is.
+        /// </summary>
+        [JsonConverter(typeof(NodeInputTypeConverter))]
+        public NodeInputTypes Type { get; set; }
+        /// <summary>
+        /// The value of the input when the graph was saved.
+        /// </summary>
+        public string Value { get; set; }
+
+        //optional properties, might be null
+        /// <summary>
+        /// If this input is a selection type a list of choices a user can select.
+        /// </summary>
+        public List<string> Choices { get; set; }
+        /// <summary>
+        /// if this input is a Number, the max value of that number.
+        /// </summary>
+        public double? MaximumValue { get; set; }
+        /// <summary>
+        /// if this input is a Number, the min value of that number.
+        /// </summary>
+        public double? MinimumValue { get; set; }
+        /// <summary>
+        /// if this input is a Number, the step value of that number when it acts as a slider.
+        /// </summary>
+        public double? StepValue { get; set; }
+        /// <summary>
+        /// if this input is a Number, the number type Double or Integer that the number returns.
+        /// </summary>
+        public string NumberType { get; set; }
```

Comment:
Similarly we can put a `DefaultValueAttribute` on this so we don't need to write out "Double".

http://www.newtonsoft.com/json/help/html/DefaultValueAttributeIgnore.htm